crates/libcontainer/src/syscall/linux.rs

```diff
@@ -334,6 +362,37 @@ impl Syscall for LinuxSyscall {
             Err(e) => bail!(e),
         }
     }
+
+    fn mount_setattr(
+        &self,
+        dirfd: RawFd,
+        pathname: &Path,
+        flags: u32,
+        mount_attr: &MountAttr,
+        size: libc::size_t,
+    ) -> Result<()> {
+        let path_pathbuf = pathname.to_path_buf();
+        let path_str = path_pathbuf.to_str();
+        let path_c_string = match path_str {
+            Some(path_str) => CString::new(path_str)?,
+            None => bail!("Invalid filename"),
+        };
+        let result = unsafe {
+            syscall!(
+                Sysno::mount_setattr,
+                dirfd,
+                path_c_string.as_ptr(),
+                flags,
+                mount_attr as *const MountAttr,
+                size
+            )
+        };
```

Comment:
I know nix/libc crate hasn't supported this system call yet. May I ask you to add TODO comment about it?
And how about creating an issue for libc/nix crate and putting the link about the issue?

Reply:
FYI: https://github.com/containers/youki/issues/987

Reply:
fixed